src/pages/store/cart.tsx

```diff
@@ -0,0 +1,336 @@
+                  <Typography variant="h5/medium" component="span">
+                    Go to Store
+                  </Typography>
+                </button>
+              </Link>
+            </div>
+          )}
+        </div>
+
+        {/* Event Picker */}
+        {((cart.length > 0 && credits >= orderTotal) || cartState === CartState.CONFIRMED) && (
+          <div className={`${styles.cartCard} ${styles.eventPicker}`}>
+            <div className={styles.header}>
+              <Typography variant="h4/bold" component="h2">
+                {cartState !== CartState.CONFIRMED ? 'Choose Pickup Event' : 'Pickup Event Details'}
+              </Typography>
+            </div>
+            <PickupEventPicker
+              events={pickupEvents.map(event => event.linkedEvent)}
+              eventIndex={pickupIndex}
+              setEventIndex={setPickupIndex}
+              active={cartState !== CartState.CONFIRMED}
+            />
+          </div>
+        )}
+
+        {/* Points Summary */}
+        <div className={`${styles.cartCard} ${styles.pointsCard}`}>
+          <Typography variant="h4/bold" component="h2" className={styles.header}>
+            {cartState !== CartState.CONFIRMED ? 'Membership Points' : 'Balance Details'}
+          </Typography>
+          <div className={styles.points}>
+            <div>
+              <Typography variant="h5/bold" component="p">
+                {cartState !== CartState.CONFIRMED ? 'Current' : 'Previous'} Balance
+              </Typography>
+              <Diamonds count={credits} />
+            </div>
+            <div>
+              <Typography variant="h5/bold" component="p">
+                Order Total
+              </Typography>
+              <div>
+                - <Diamonds count={orderTotal} />
+              </div>
+            </div>
+            <hr />
+            <div>
+              <Typography variant="h5/bold" component="p">
+                Remaining Balance
+              </Typography>
+              <Diamonds count={credits - orderTotal} />
+            </div>
+          </div>
+        </div>
+
+        {credits < orderTotal && (
+          <Typography variant="h5/regular" component="p" className={styles.warning}>
+            You do not have enough membership points to checkout. Earn more membership points by
+            attending ACM events!
+          </Typography>
+        )}
+      </div>
+    </div>
+  );
+};
+
+export default StoreCartPage;
+
+const getServerSidePropsFunc: GetServerSideProps = async ({ req, res }) => {
+  const AUTH_TOKEN = getServerCookie(CookieType.ACCESS_TOKEN, { req, res });
+
+  // recover saved items and reset cart to [] if it's invalid
+  let savedItems;
+  try {
+    const cartCookie = getServerCookie(CookieType.CART, { req, res });
+    savedItems = JSON.parse(cartCookie);
+    if (!Array.isArray(savedItems)) throw new Error();
+  } catch {
+    savedItems = [];
+  }
+
+  const savedCartPromises = Promise.all(
```

Comment:
```suggestion
  const savedCart = Promise.all(
```
nit to rename the variable because technically this isn't returning a promise array since you chained the `.then()` on it

Reply:
Doesn't `.then()` return another promise? My understanding is since nothing has been awaited, no promises have been resolved, and API requests haven't been made until savedCartPromises is awaited with the others below.

Reply:
Oop yea I think you're right